icechunk/src/bin/icechunk/main.rs

```diff
@@ -0,0 +1,8 @@
+use clap::Parser;
+use icechunk::cli::interface::{run_cli, IcechunkCLI};
+
+#[tokio::main]
+async fn main() {
+    let cli = IcechunkCLI::parse();
+    let _ = run_cli(cli).await;
```

Comment:
This should probably return an error, something like:
```rust
fn main() -> Result<(), Box<dyn Error>>
```
we will want the status code to be != 0 if some error happened

Reply:
Improved error handling in https://github.com/earth-mover/icechunk/pull/716/commits/1724fcd2fabb4b53d17382693e9ba30bc623d9ab